MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2443,7 +2443,13 @@ public string GetHardwareAcceleratedVideoDecoder(VideoType videoType, MediaStrea
                     {
                         case "avc":
                         case "h264":
-                            if (_mediaEncoder.SupportsDecoder("h264_cuvid") && encodingOptions.HardwareDecodingCodecs.Contains("h264", StringComparer.OrdinalIgnoreCase))
+                            // cuvid decoder does not support >8 bits on h264, fall back onto h264.
+                            if (videoStream.BitDepth.HasValue && videoStream.BitDepth.Value > 8)
+                            {
+                                encodingOptions.HardwareDecodingCodecs = Array.Empty<string>();
```

Comment:
This leaves a taste of a hack... Usually functions named `GetFoo()` are not modifying the world. Can we do it in a more elegant way somehow?

Reply:
It would probably easiest to rename the `GetHardwareAcceleratedVideoDecoder` to `AdjustHardwareAccelerationParameters` , because this exact hack is done for in this same function for h264_qsv (quicksync) and the logic is pretty ingrained, and would be more-so if we had to eventually achieve software fallback.  

What would you suggest?

Reply:
I like renaming to `AdjustFoo()`, go ahead